Source/include/Ably/ARTPush.h

```diff
@@ -29,7 +29,12 @@ NS_ASSUME_NONNULL_BEGIN
 @optional
 
 /**
- Ably will call the implementation of this method when the registration process is completed with failure.
+ Ably will call the implementation of this method when the token update process is completed with success or with failure.
+ */
+- (void)didUpdateAblyPush:(nullable ARTErrorInfo *)error;
+
+/**
+ Same as `didUpdateAblyPush:`, but called only in case of failure for backward compatibility.
```

Comment:
I think we need to explain these two methods better — it's a bit confusing because they have identical signatures. What does it mean if Ably calls `didAblyPushRegistrationFail:(nullable ARTErrorInfo *)error;` with a `nil` error, compared to if `- (void)didUpdateAblyPush:(nullable ARTErrorInfo *)error;` does the same? Is it even possible for the former to be called with a `nil` error?

Reply:
I think that `didAblyPushRegistrationFail:` should be removed from the specs (https://sdk.ably.com/builds/ably/specification/main/features/#RSH3e3a). Since some platforms (apple for now) need update its token with a success callback.

Reply:
>Is it even possible for the former to be called with a nil error?

Library doesn't do that ([RSH3e3a)](https://sdk.ably.com/builds/ably/specification/main/features/#RSH3e)).

Reply:
I've created a PR with the issue for that in specs repo - https://github.com/ably/specification/pull/160

Reply:
https://github.com/ably/specification/pull/160#discussion_r1269689397

Reply:
>Created https://github.com/ably/ably-cocoa/issues/1784 for making that error non-nullable

Should I address this in a different PR? @lawrence-forooghian

Reply:
> > Created #1784 for making that error non-nullable
> 
> Should I address this in a different PR? @lawrence-forooghian

Yeah, it's not related to this one, nor do I think it's urgent